.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,14 @@
+on:
+  pull_request:
+  push:
+    branches: [main]
```

Comment:
add master as well

Reply:
btw can you also change the default python version from python3.7 to just python3 in `.pre-commit-config.yaml`

I think this was the issue that pre-commit requires only 3.7, but it doesn't have to

Reply:
Done. Was facing Some issue for test pre-commit. But after `pre-commit clean` & `export PIP_DEFAULT_TIMEOUT=100` command it is working on my machine.